src/segment_language.go

```diff
@@ -166,13 +168,22 @@ func (l *language) hasLanguageFiles() bool {
 // setVersion parses the version string returned by the command
 func (l *language) setVersion() error {
 	for _, command := range l.commands {
-		if !l.env.hasCommand(command.executable) {
-			continue
-		}
-		versionStr, err := l.env.runCommand(command.executable, command.args...)
-		if exitErr, ok := err.(*commandError); ok {
-			l.exitCode = exitErr.exitCode
-			return fmt.Errorf("err executing %s with %s", command.executable, command.args)
+		var versionStr string
+		var err error
+		if command.getVersion == nil {
+			if !l.env.hasCommand(command.executable) {
+				continue
+			}
+			versionStr, err = l.env.runCommand(command.executable, command.args...)
+			if exitErr, ok := err.(*commandError); ok {
+				l.exitCode = exitErr.exitCode
+				return fmt.Errorf("err executing %s with %s", command.executable, command.args)
+			}
+		} else {
```

Comment:
if we swap the if, we can simply return early or continue in case we have an error

Reply:
I don't really get it. If `getVersion` is not null we use it to read the version, if not we use the command. Both will output a `versionStr` that needs to be parsed using the regex. I probably missed your point.
![image](https://user-images.githubusercontent.com/1829553/145354121-a9564885-0a54-40d0-9be5-d77485a7aa7a.png)

Reply:
you're right.